components/config-mgmt-service/integration_test/event_task_counts_test.go

```diff
@@ -164,6 +166,561 @@ func TestEventTaskCountsCountOnlyFilteredChefServers(t *testing.T) {
+			description: "Action with several projects where one matches one of several requested project allowed",
+			ctx:         contextWithProjects([]string{"project3", "project10", "project12", "project13"}),
+			actions: []iBackend.InternalChefAction{
+				{
+					Task:     "create",
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			expected: &response.EventCounts{
+				Total: 1,
+				Counts: []*response.EventCount{
+					{
+						Name:  "create",
+						Count: 1,
+					},
+				},
+			},
+		},
+		{
+			description: "Two Actions with several projects where one matches one of several requested project allowed",
+			ctx:         contextWithProjects([]string{"project3", "project10", "project12", "project13"}),
+			actions: []iBackend.InternalChefAction{
+				{
+					Task:     "create",
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+				{
+					Task:     "delete",
+					Projects: []string{"project13", "project14", "project17", "project16"},
+				},
+			},
+			expected: &response.EventCounts{
+				Total: 2,
+				Counts: []*response.EventCount{
+					{
+						Name:  "delete",
+						Count: 1,
+					},
+					{
+						Name:  "create",
+						Count: 1,
+					},
+				},
+			},
+		},
+		{
+			description: "Action with several projects where none matches several requested project allowed",
+			ctx:         contextWithProjects([]string{"project14", "project10", "project12", "project13"}),
+			actions: []iBackend.InternalChefAction{
+				{
+					Task:     "create",
+					Projects: []string{"project3", "project4", "project7", "project6"},
+				},
+			},
+			expected: &response.EventCounts{},
+		},
+		{
+			description: "Action with several projects where two matches two of several requested project allowed",
+			ctx:         contextWithProjects([]string{"project3", "project10", "project12", "project13"}),
+			actions: []iBackend.InternalChefAction{
+				{
+					Task:     "create",
+					Projects: []string{"project3", "project10", "project7", "project6"},
+				},
+			},
+			expected: &response.EventCounts{
+				Total: 1,
+				Counts: []*response.EventCount{
+					{
+						Name:  "create",
+						Count: 1,
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(fmt.Sprintf("Project filter: %s", test.description), func(t *testing.T) {
+			for index := range test.actions {
+				test.actions[index].RecordedAt = time.Now()
+				test.actions[index].Id = newUUID()
+			}
+
+			suite.IngestActions(test.actions)
+			defer suite.DeleteAllDocuments()
+
+			res, err := cfgmgmt.GetEventTaskCounts(test.ctx, &request.EventCountsFilter{})
+			assert.NoError(t, err)
+
+			// test response
+			assert.Equal(t, test.expected.Total, res.Total)
+
+			for _, expectedCount := range test.expected.Counts {
+				matchingResCount, err := findEventCountByName(expectedCount.Name, res.Counts)
+				assert.NoError(t, err)
+
+				assert.Equal(t, expectedCount.Count, matchingResCount.Count)
+			}
+		})
```

Comment:
I'm a little confused why the test cases build up a `response.EventCounts` struct and then only refer to its properties and not the struct itself in the test. That seems to couple the test implementation to the code's implementation unnecessarily.

Reply:
Also, because of how the test is structured you could have a passing test where the implementation included more information than was included in the expected data in the test case. Was that intentional?

Reply:
For the second comment. I only see that is the case if there are two counts for the same type returned. I think that should be tested somewhere else.

Reply:
For the second comment. I used the `response.EventCounts` because it matches the response closer. I would have like to have `assert.Equal(t, test.expected.Counts, res)` but I wanted to account for the `Counts` being in a different order.

Reply:
Is the order significant? Does `ElementsMatch` work for these?

Reply:
The order is not significant. I just assumed that `ElementsMatch` would not work because the `Count` is an array of structs, but now I want to test it.

Reply:
@phiggins It does work with `ElementsMatch`! Updating now. Thanks.

Reply:
Thank you for humoring my pedantry 😅